scripts/test_backref_cost_attribution_v3_protocol.py

```
@@ -0,0 +1,168 @@
+        if text[index] == "{":
+            depth += 1
+        elif text[index] == "}":
+            depth -= 1
+            if depth == 0:
+                return text[brace:index + 1]
+    raise AssertionError(f"unterminated function: {name}")
+
+
+def semantic_identity() -> None:
+    frozen = subprocess.check_output(
+        ["git", "show", "673bc3fa:src/enc/backward_references_cost_enc.c"],
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)